her-tech-trail/Trail-backend
# #2 Add signup/signup Feature

#### Description of Task to be completed?
- add signup implementation 
- add signin implementation
- add user model
- update eslintrc rules
- add user route

 #### How should this be manually tested?
- clone this repository
- run `npm install` to download the related dependencies
- run `npm run migrate` to migrate models to database
- run `npm run seed` to ensure migrate was successful
- run `npm run dev` to start the development server
- open postman and send POST request to the url `localhost:5000/api/v1/user/signup`
- provide the following information 
```json
{
	"email": "kelechioliver96@gmail.com",
	"password": "password",
	"firstName": "John",
	"lastName": "Doe"
}
```
- you shoud see a successful response code of `201`

 #### Any background context you want to provide?
- ensure you have your database setup

 #### What are the relevant pivotal tracker stories?
- none, implement signup signin

 #### Screenshots (if appropriate)
- none

 #### Questions:
- none
